### ISSUE TITLE
Bug with Null safety with Flutter 2.0.1 

### ISSUE THREAD

**Issue by SmarterVision** [fysoul17/google_maps_place_picker#107]
**Describe the bug**
Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.
      prediction.placeId,
Error: Method 'substring' cannot be called on 'String?' because it is potentially null.
Try calling using ?. instead.
Error: Method 'substring' cannot be called on 'String?' because it is potentially null.
Try calling using ?. instead.
Error: Property 'length' cannot be accessed on 'String?' because it is potentially null.
Try accessing using ?. instead.
Error: Method 'substring' cannot be called on 'String?' because it is potentially null.
Try calling using ?. instead.

**To Reproduce**
When you run the app you will get the errors above

**Expected behavior**
```
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_place_picker-2.0.0-nullsafety.1/lib/src/place_picker.dart:320:18: Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.
      prediction.placeId,
                 ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_place_picker-2.0.0-nullsafety.1/lib/src/components/prediction_tile.dart:37:42: Error: Method 'substring' cannot be called on 'String?' because it is potentially null.
Try calling using ?. instead.
            text: prediction.description.substring(0, matchedSubString.offset as int?),
                                         ^^^^^^^^^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_place_picker-2.0.0-nullsafety.1/lib/src/components/prediction_tile.dart:46:40: Error: Method 'substring' cannot be called on 'String?' because it is potentially null.
Try calling using ?. instead.
          text: prediction.description.substring(matchedSubString.offset as int, matchedSubString.offset + matchedSubString.length as int?),
                                       ^^^^^^^^^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_place_picker-2.0.0-nullsafety.1/lib/src/components/prediction_tile.dart:52:86: Error: Property 'length' cannot be accessed on 'String?' because it is potentially null.
Try accessing using ?. instead.
      if (matchedSubString.offset + matchedSubString.length < prediction.description.length) {
                                                                                     ^^^^^^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_place_picker-2.0.0-nullsafety.1/lib/src/components/prediction_tile.dart:55:42: Error: Method 'substring' cannot be called on 'String?' because it is potentially null.
Try calling using ?. instead.
            text: prediction.description.substring(matchedSubString.offset + matchedSubString.length as int),
                                         ^^^^^^^^^
```

**Flutter Doctor -v**
Flutter 2.0.1 Stable Release

**Comment by kirkcharlesniv** [comment]
Can confirm, currently experiencing this issue too.

**Comment by kirkcharlesniv** [comment]
Search is not working too, at `_pickPrediction()`:`place_picker.dart` there's an error on: 

```
 await provider!.places.getDetailsByPlaceId(
      prediction.placeId!, // Error here, Unhandled Exception: type 'Null' is not a subtype of type 'String' in type cast
      sessionToken: provider!.sessionToken,
      language: widget.autocompleteLanguage,
    );
```

**Comment by SmarterVision** [comment]
> Can confirm, currently experiencing this issue too.

# In your pubspec.yaml ( just a temporary solution)

  google_maps_place_picker:
    git:
      url: git://github.com/SmarterVision/google_maps_place_picker.git

**Comment by manpreetpatil** [comment]
I'm also facing same issue

**Comment by fysoul17** [comment]
Fixed at version 2.0.0-nullsafety.2

**Comment by kirkcharlesniv** [comment]
Please upgrade to `google_maps_webservice: 0.0.20-nullsafety.2` too for search fix. Thanks!

https://github.com/fysoul17/google_maps_place_picker/issues/100#issuecomment-798925279

**Comment by fysoul17** [comment]
@kirkcharlesniv 
Available at version 2.0.0-nullsafety.3
